src/DirectusProvider.tsx

```
@@ -144,14 +144,14 @@ export const DirectusProvider = <T extends TypeMap = TypeMap>({
           })) as UserType;
 
           if (dUser) {
-            newAuthState = 'authenticated';
+            newAuthState = AuthStates.AUTHENTICATED;
             setUser(dUser);
           }
         }
       } catch (error) {
         console.log('auth-error', error);
```

Can we print an useful message here?

---

That's not realy neccessery, we could even drop the console log. The only time an error ocures there is if autoLogin is set to true but no valide auth token is stored.

---

Are you suggesting to remove the try/catch and just call directus API right?

---

No, I would sugest to remove the console.log from the catch.

---

So any error would be ignored?

---

Yes

---

Let's solve this before merge. If I understand correcly, with `autoLogin` enabled (which isn't documented at all) that code is executed, calling Directus refresh is called. If something go wrong we don't provide any message and/or log nothing?

I'm trying to also understand this https://docs.directus.io/reference/old-sdk.html#basic-authentication ... where the token come from? This is also undocumented I think...

---

You are right, I totally forgot to add the `autoLogin` feature to the README.

The feature works as follows:
When a user logs in, the Directus SDK saves an auth token returned by the API. The token needs to be refreshed at defined intervals, and that's why the call to `directus.auth.token` has to be async, as sometimes the token needs to be refreshed first.

If a user was logged in during a previous session, the refresh token is still saved. When `autoLogin` is enabled, the provider checks if the refresh token is still valid using `directus.auth.refresh()`. If the refresh is successful and a new auth token is issued, the user becomes `authenticated`.

I apologize for the oversight, and I will add the necessary information about the `autoLogin` feature to the README.

Thank you for bringing this to my attention, and please let me know if you have any other feedback or questions.

---

Thank you for your work and the explanation, now I understand. The question how to handle the `refresh` call correcly remains (could be a simple network error).

---

We could include a onAutoLoginError funktion?

---

I'm not a React guru 😄  and I can't really answer to that question, in addition it's an async errors so it would be impossibile to catch with something like https://github.com/bvaughn/react-error-boundary.

Adding `onAutoLoginError` prop seems appropriate as user can handle it as he wish.